test/Kestrel.FunctionalTests/HttpsConnectionAdapterTests.cs

```diff
@@ -381,7 +382,44 @@ public void ThrowsWhenNoServerCertificateIsProvided()
                 }
             }
         }
+#if NETCOREAPP2_2
+        [Fact]
+        public async Task CertificatePassedToHttpContextWhenHttp1and2Enabled()
+        {
+            var listenOptions = new ListenOptions(new IPEndPoint(IPAddress.Loopback, 0))
+            {
+                Protocols = HttpProtocols.Http1AndHttp2,
+                ConnectionAdapters =
+                {
+                    new HttpsConnectionAdapter(new HttpsConnectionAdapterOptions
+                    {
+                        ServerCertificate = _x509Certificate2,
+                        ClientCertificateMode = ClientCertificateMode.RequireCertificate,
+                        ClientCertificateValidation = (certificate, chain, sslPolicyErrors) => true
+                    })
+                }
+            };
 
+            using (var server = new TestServer(context =>
+            {
+                var tlsFeature = context.Features.Get<ITlsConnectionFeature>();
+                Assert.NotNull(tlsFeature);
+                Assert.NotNull(tlsFeature.ClientCertificate);
+                Assert.NotNull(context.Connection.ClientCertificate);
+                return context.Response.WriteAsync("hello world");
+            }, new TestServiceContext(LoggerFactory), listenOptions))
+            {
+                using (var client = new TcpClient())
+                {
+                    // SslStream is used to ensure the certificate is actually passed to the server
+                    // HttpClient might not send the certificate because it is invalid or it doesn't match any
+                    // of the certificate authorities sent by the server in the SSL handshake.
```

Comment:
I don't understand this comment, the other tests in this class use HttpClientSlim which has a validateCertificate option.

Reply:
Consolidated the test code